docs/modeling/units.rst

```diff
@@ -0,0 +1,279 @@
+********************************
+Support for units and quantities
+********************************
+
+.. note:: The functionality presented here was recently added. If you run into
+          any issues, please don't hesitate to open an issue in the `issue
+          tracker <https://github.com/astropy/astropy/issues>`_.
+
+The `astropy.modeling` package includes partial support for the use of units and
+quantities in model parameters, models, and during fitting. At this time, only
+some of the built-in models (such as
+:class:`~astropy.modeling.functional_models.Gaussian1D`) support units, but this
+will be extended in future to all models where this is appropriate.
+
+Setting parameters to quantities
+================================
+
+Models can take :class:`~astropy.units.Quantity` objects as parameters::
+
+    >>> from astropy import units as u
+    >>> from astropy.modeling.models import Gaussian1D
+    >>> g1 = Gaussian1D(mean=3 * u.m, stddev=2 * u.cm, amplitude=3 * u.Jy)
+
+Accessing the parameter then returns a Parameter object that contains the value
+and the unit::
+
+    >>> g1.mean
+    Parameter('mean', value=3.0, unit=m)
+
+It is then possible to access the individual properties of the parameter::
+
+    >>> g1.mean.name
+    'mean'
+    >>> g1.mean.value
+    3.0
+    >>> g1.mean.unit
+    Unit("m")
+
+If a parameter has been initialized as a Quantity, it should always be set to a
+quantity, but the units don't have to be compatible with the initial ones::
+
+    >>> g1.mean = 3 * u.s
+    >>> g1
+    <Gaussian1D(amplitude=3.0 Jy, mean=3.0 s, stddev=2.0 cm)>
+
+To change the value of a parameter and not the unit, simply set the value
+property::
+
+    >>> g1.mean.value = 2
+    >>> g1
+    <Gaussian1D(amplitude=3.0 Jy, mean=2.0 s, stddev=2.0 cm)>
+
+Setting a parameter which was originally set to a quantity to a scalar doesn't
+work because it's ambiguous whether the user means to change just the value and
+preserve the unit, or get rid of the unit::
+
+    >>> g1.mean = 2  # doctest: +IGNORE_EXCEPTION_DETAIL
+    Traceback (most recent call last):
+    ...
+    UnitsError : The 'mean' parameter should be given as a Quantity because it
+    was originally initialized as a Quantity
+
+On the other hand, if a parameter previously defined without units is given a
+Quantity with a unit, this works because it is unambiguous::
+
+    >>> g0 = Gaussian1D(mean=3)
```

Comment:
nitpicking, but the name g0 is unintuitive when all the rest of the names are ordered, looking in the middle of the docs, I assumed it was already been used before g1